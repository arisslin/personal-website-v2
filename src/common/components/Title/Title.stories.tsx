import * as React from 'react';
import { Meta, Story } from '@storybook/react';
import TitleComponent from './Title';

export default {
  title: 'Components/Title',
  component: TitleComponent,
  parameters: {
    docs: {
      description: {
        component: getDescription(),
      },
    },
  },
} as Meta;

function getDescription(): string {
  return (
    '<h2>About</h2>' +
    '<p>' +
    'The <b>Title</b> component got developed in March 2022. It shows the' +
    'title of the entire website. It can be used on every page of the' +
    "website but it's mainly but it was designed to be displayed on the " +
    'home page.' +
    '</p>'
  );
}

export const Title: Story = () => <TitleComponent />;
