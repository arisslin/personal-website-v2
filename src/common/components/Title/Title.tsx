import * as React from 'react';
import styled from 'styled-components/macro';
import { breakpoints } from '../../styles/layout';

const Title = () => (
  <TitleStyled>
    Andreas Rissling
    <hr />
    <span>Web Developer</span>
  </TitleStyled>
);

const TitleStyled = styled.h1`
  margin-top: 50px;
  font-size: 1.75rem;
  font-weight: 300;
  letter-spacing: 3px;
  text-align: center;

  @media screen and (min-width: ${breakpoints.medium}) {
    margin-top: 70px;
    font-size: 2rem;
  }

  hr {
    margin-top: 0.8rem;
    margin-bottom: 0.5rem;
    border: none;
    height: 1px;
    width: 100px;
    background-color: var(--color-dark);

    @media screen and (min-width: ${breakpoints.medium}) {
      width: 110px;
    }
  }

  span {
    font-size: 1.5rem;

    @media screen and (min-width: ${breakpoints.medium}) {
      font-size: 1.75rem;
    }
  }
`;

export default Title;
