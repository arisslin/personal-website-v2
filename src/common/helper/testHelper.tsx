import { cleanup, fireEvent, render } from '@testing-library/react';

/**
 * Does a snapshot test for component it got.
 * @param {JSX.Element} component to creating snapshot for
 * @param {string} variant to create a postfix for the displayed test string
 * @return {void}
 */
export function doSnapshotTest(component: JSX.Element, variant?: string): void {
  const postfix = `${component.type.name as string} ${variant ? variant : ''}`;

  test(`Snapshot test for ${postfix}`, () => {
    const { container } = render(component);

    expect(container).toMatchSnapshot();
    cleanup();
  });
}

/**
 * Fires an click event with 'fireEvent' from '@testing-library/react' on an
 * element.
 * @param {HTMLElement} element to fire click on.
 * @param {boolean} bubbles attribute of 'MouseEvent'.
 * @param {boolean} cancelable attribute of 'MouseEvent'.
 * @return {void}
 */
export function fireClickEvent(
  element: Element,
  bubbles: boolean,
  cancelable: boolean
) {
  fireEvent(element, new MouseEvent('click', { bubbles, cancelable }));
}
