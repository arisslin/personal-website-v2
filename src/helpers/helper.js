export const getText = (wantText) => {
  if (typeof wantText === 'boolean') {
    return wantText ? 'text' : null;
  }

  return null;
};
