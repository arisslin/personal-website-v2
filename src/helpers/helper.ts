export const getText = (wantText: boolean) => (wantText ? 'text' : null);
