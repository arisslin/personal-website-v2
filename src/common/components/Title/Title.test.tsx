import * as React from 'react';
import { render, screen } from '@testing-library/react';
import Title from './Title';
import { doSnapshotTest } from '../../helper/testHelper';
import 'jest-styled-components';

describe('Title', () => {
  it('should render with main and sub title', () => {
    render(<Title />);
    expect(screen.getByText('Andreas Rissling')).toBeInTheDocument();
    expect(screen.getByText('Web Developer')).toBeInTheDocument();
  });

  doSnapshotTest(<Title />, 'default');
});
