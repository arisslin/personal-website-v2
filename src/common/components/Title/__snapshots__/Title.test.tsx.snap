// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Snapshot test for Title default 1`] = `
.c0 {
  margin-top: 50px;
  font-size: 1.75rem;
  font-weight: 300;
  -webkit-letter-spacing: 3px;
  -moz-letter-spacing: 3px;
  -ms-letter-spacing: 3px;
  letter-spacing: 3px;
  text-align: center;
}

.c0 hr {
  margin-top: 0.8rem;
  margin-bottom: 0.5rem;
  border: none;
  height: 1px;
  width: 100px;
  background-color: var(--color-dark);
}

.c0 span {
  font-size: 1.5rem;
}

@media screen and (min-width:768px) {
  .c0 {
    margin-top: 70px;
    font-size: 2rem;
  }
}

@media screen and (min-width:768px) {
  .c0 hr {
    width: 110px;
  }
}

@media screen and (min-width:768px) {
  .c0 span {
    font-size: 1.75rem;
  }
}

<div>
  <h1
    class="c0"
  >
    Andreas Rissling
    <hr />
    <span>
      Web Developer
    </span>
  </h1>
</div>
`;
